lib/xtool/xtcore.cpp

```diff
@@ -142,7 +142,8 @@ int main(int argc, char *argv[])
         } else if (cmd_key == "-port") {
             if (argc > i) {
                 i++;
-                autoconnectPort = strtol(argv[i], &argv[i], 0);
+                char *err;
+                autoconnectPort = (unsigned short)strtol(argv[i], &err, 0);

```

Comment:
Second argument of `strtol` is not describe an error. It's should pointer that will be points to the next byte of the last char of the converted number. The argument may be equals to `NULL` if  we no needs know last converted symbol.

I guess the code should looks like `strtol(argv[i], NULL, 0)`

Reply:
"It's should be a pointer that will point to the next byte ". I know thus why I wanted to avoid reusing argv[i] but fair point here we re not reusing it here.